### PR TITLE
feat: Keycloak integration

### PR DESCRIPTION
**Pull request by wtrocki**
Added login/logout/getProfile capacity

screen shot 2019-01-10 at 5 19 21 pm

Sync integration:
screen shot 2019-01-10 at 5 28 18 pm

Outstanding - while this work js-sdk sync package needs some refactoring as api do not allow to refresh token properly. Going to work on this

### CODE REVIEW

**Comment by wtrocki** [comment]
@darahayes maybe we can have both server and client in the single branch to we could verify that together?

File: src/app/pages/home/home.page.html [hunk 1]
```diff
@@ -5,6 +5,12 @@
       <ion-button (click)="openNewItemPage()">
         <ion-icon slot="icon-only" name="add"></ion-icon>
       </ion-button>
+      <div *ngIf="auth.authService">
```
Comment (wtrocki) [inline src/app/pages/home/home.page.html]: This is supper ugly but I'm not designer.

File: src/app/pages/home/home.page.ts [hunk 2]
```diff
@@ -34,6 +37,11 @@ export class HomePage implements OnInit {
     // Local cache can be updated by mutations that happen on the app
     await this.loadData();
     await this.setupQueueStatusBar();
+    this.auth.init().then(() => {
+      this.auth.authService.extract().loadUserProfile().success((profile) => {
+        this.user = profile.username;
+      });
+    });
```
Comment (wtrocki) [inline src/app/pages/home/home.page.ts]: Note: Add catch.

**Comment by wei-lee** [comment]
Brilliant!! 🥇

File: src/app/services/voyager.service.ts [hunk 3]
```diff
@@ -71,3 +74,20 @@ export class VoyagerService {
...
 }
 
+class AuthHeaderProvider implements HeaderProvider {
+
+  constructor(private auth: AuthService) {
+  }
+  getHeaders(): { [index: string]: string; } {
+    if (this.auth.authService) {
+      // TODO async
+      this.auth.authService.extract().updateToken(30);
```
Comment (wtrocki) [inline src/app/services/voyager.service.ts]: Move to async

File: src/mobile-services.js [hunk 4]
```diff
@@ -4,6 +4,27 @@ var config = {
   "namespace": "myproject",
   "clientId": "app-js",
   "services": [
+    {
```
Comment (wtrocki) [inline src/mobile-services.js]: Remove this - it's only for verification purposes.

**Comment by wtrocki** [comment]
Review/approval needed.

**Pull request merged**